### PR TITLE
Use QgsMessageBar instead of QMessageBoxes to show result of testing connection

### PR DESCRIPTION
**Pull request by alexbruy**
## Description
All dialogs for adding new database connection, except MSSQL, use `QMessageBox` to report results of the connection test (success/failure). This PR replaces them with internal `QgsMessageBar`.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/providers/db2/qgsdb2newconnection.cpp [hunk 1]
```diff
@@ -147,7 +145,7 @@ void QgsDb2NewConnection::on_cb_trustedConnection_clicked()
 
 QgsDb2NewConnection::~QgsDb2NewConnection()
 {
-
+  delete bar;
```
Comment (m-kuhn) [inline src/providers/db2/qgsdb2newconnection.cpp]: Where is this created? In the .ui file?
Normally there's no need for manual cleanup with widgets if they are properly parented.

**Comment by m-kuhn** [comment]
Yay, less dialogs :)

File: src/providers/db2/qgsdb2newconnection.cpp (hunk 1 above)
Comment (alexbruy) [inline src/providers/db2/qgsdb2newconnection.cpp]: Yes, in the `.ui` file. I will remove it

**Comment by nyalldawson** [comment]
Nice!

**Pull request merged**